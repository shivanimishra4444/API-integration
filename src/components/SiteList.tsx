import * as React from 'react'
import { fetchData } from '../model/Model'
import SiteListItem from './SiteListItem'

interface IState {
  sites: []
}
export default class SiteList extends React.Component<any, IState> {
  constructor(props) {
    super(props)
    this.state = { sites: [] }
  }
  public async componentDidMount() {
    const sites = await fetchData()
    this.setState({ sites })
  }
  public render() {
    if (!this.state.sites) {
      return
    }
    return (
      <div>
        {this.state.sites.map((site: { name: string; id: number }) => (
          <SiteListItem key={site.id} site={site} />
        ))}
      </div>
    )
  }
}
