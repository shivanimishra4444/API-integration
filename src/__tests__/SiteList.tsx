import * as React from 'react'
import SiteList from '../components/SiteList'
import * as Adapter from 'enzyme-adapter-react-16'
import { configure, mount } from 'enzyme'
configure({ adapter: new Adapter() })

describe('<SiteList />', () => {
  const wrapper = mount(<SiteList />)

  it('matches snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })
  it('render 1 <SiteList> component snapshot', () => {
    expect(wrapper).toHaveLength(1)
  })
})
