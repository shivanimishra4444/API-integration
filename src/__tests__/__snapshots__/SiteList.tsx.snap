// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SiteList /> matches snapshot 1`] = `ReactWrapper {}`;
